internal/commands/pull/inputs.go

```diff
@@ -107,5 +115,73 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 		return realm.App{}, err
 	}
 
+	// TODO(REALMC-9462): remove this once /apps has "template_id" in the payload
+	app, err = clients.Realm.FindApp(app.GroupID, app.ID)
+	if err != nil {
+		return realm.App{}, err
+	}
+
 	return app, nil
 }
+
+func (i *inputs) resolveClientTemplates(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, error) {
+	compatibleTemplates, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, err
+	}
+
+	if i.TemplateID != "" {
+		for _, template := range compatibleTemplates {
+			if template.ID == i.TemplateID {
+				templateZip, ok, err := realmClient.ClientTemplate(groupID, appID, template.ID)
+				if err != nil {
+					return nil, err
+				} else if !ok {
+					return nil, fmt.Errorf("template '%s' does not have a frontend to pull", template.ID)
+				}
+				return map[string]*zip.Reader{template.ID: templateZip}, nil
+			}
+		}
+		return nil, fmt.Errorf("template '%s' is not compatible with this app", i.TemplateID)
+	}
+
+	if len(compatibleTemplates) == 0 {
+		return nil, nil
+	}
+
+	// TODO(REALMC-9474) fix to align the template selection flow with app create
+	if proceed, err := ui.Confirm("Would you like to export with a template?"); err != nil {
+		return nil, err
+	} else if !proceed {
+		return nil, nil
+	}
+
+	nameOptions := make([]string, len(compatibleTemplates))
+	for idx, compatibleTemplate := range compatibleTemplates {
+		nameOptions[idx] = fmt.Sprintf("[%s]: %s", compatibleTemplate.ID, compatibleTemplate.Name)
+	}
+
+	var selectedTemplateIdxs []int
+	if err := ui.AskOne(
+		&selectedTemplateIdxs,
+		&survey.MultiSelect{Message: "Which template(s) would you like to export this app with", Options: nameOptions}); err != nil {
+		return nil, err
+	}
+
+	templateIDs := make([]string, len(selectedTemplateIdxs))
+	for idx, selectedTemplateIdx := range selectedTemplateIdxs {
+		templateIDs[idx] = compatibleTemplates[selectedTemplateIdx].ID
```

Comment:
[question] is the double `for` loop necessary here?  could we just create the `result` map inside of this `range selectedTemplateIdxs`?